atc/exec/retry_error_step.go

```diff
@@ -0,0 +1,53 @@
+package exec
+
+import (
+	"context"
+	"reflect"
+	"regexp"
+
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/lager/lagerctx"
+)
+
+type Retriable struct{}
+
+func (r Retriable) Error() string {
+	return "retriable"
+}
+
+type RetryErrorStep struct {
+	Step
+
+	delegate LogErrorStepDelegate
+}
+
+func RetryError(step Step, delegate LogErrorStepDelegate) Step {
+	return RetryErrorStep{
+		Step:     LogError(step, delegate),
+		delegate: delegate,
+	}
+}
+
+func (step RetryErrorStep) Run(ctx context.Context, state RunState) error {
+	logger := lagerctx.FromContext(ctx)
+
+	runErr := step.Step.Run(ctx, state)
```

Comment:
This is fine but took me a little bit of time to grasp, and I'm wondering if others who are unfamiliar with this part of the codebase will feel the same. 

Maybe the indirection introduced by having `RetryError` wrap `LogError` (which was itself wrapping a `Step`) is not necessary.  One alternative could be to move the essential logic of `RetryError` - conditionally setting error type to "Retriable" and returning that error - to `LogError` instead.

Another alternative is to check if the error message has "worker  .... disappeared" in the section of `engine.go` where you're checking if the error is of type `Retriable`.

I can maybe guess some reasons why those alternatives aren't as good and would like to know your thoughts.